### PR TITLE
chore(hardware-trezor): refactor TrezorKeyAgent to use Transform interface

### PR DESCRIPTION
**Pull request by tomislavhoracek**
This PR introduces a TrezorKeyAgent Transform interface refactor

# Context

as per Slack: https://input-output-rnd.slack.com/archives/C05GY0FSW73/p1707252256148049

### CODE REVIEW

File: packages/hardware-trezor/src/transformers/keyPaths.ts [hunk 1]
```diff
@@ -15,13 +15,13 @@ export const resolvePaymentKeyPathForTxIn = (
   }
 };
 
-// Resolves the stake key path for known addresses for the given reward address.
 export const resolveStakeKeyPath = (
   rewardAddress: Cardano.RewardAddress | undefined,
-  context: TrezorTxTransformerContext
+  context?: TrezorTxTransformerContext
```
Comment (mkazlauskas) [inline packages/hardware-trezor/src/transformers/keyPaths.ts:20]: It is not possible to resolve stake key path without context, so this change seems a little bit odd

File: packages/hardware-trezor/src/transformers/tx.ts [hunk 2]
```diff
@@ -1,50 +1,38 @@
...
-      ? mapRequiredSigners(body.requiredExtraSignatures, context)
-      : undefined,
-    totalCollateral: body.totalCollateral ? body.totalCollateral.toString() : undefined,
-    ttl: body.validityInterval?.invalidHereafter?.toString(),
-    validityIntervalStart: body.validityInterval?.invalidBefore?.toString(),
-    withdrawals: mapWithdrawals(body.withdrawals ?? [], context)
-  };
+export const trezorTxTransformer: Transformer<
+  Cardano.TxBody,
+  Omit<Trezor.CardanoSignTransaction, 'signingMode' | 'derivationType' | 'includeNetworkId'>,
```
Comment (mkazlauskas) [inline packages/hardware-trezor/src/transformers/tx.ts:11]: Are those properties never needed? :thinking:

File: packages/hardware-trezor/src/transformers/keyPaths.ts (hunk 1 above)
Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/keyPaths.ts:20]: you are right, nice catch. Thanks @mkazlauskas

File: packages/hardware-trezor/src/transformers/tx.ts (hunk 2 above)
Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/tx.ts:11]: `derivationType` and `includeNetworkId` are never used. 
SigningMode is determined after data mapping.

File: packages/hardware-trezor/src/transformers/keyPaths.ts (hunk 1 above)
Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/keyPaths.ts]: @marcin-mazurek done in [bda86db](https://github.com/input-output-hk/cardano-js-sdk/pull/1096/commits/bda86dbd93a5e526a3c4d4960052ba0506583219)

p.s. Will squash it once you approve the changes

**Review by mkazlauskas — APPROVED** [review]
Nice work! :rocket:

File: packages/hardware-trezor/src/transformers/assets.ts [hunk 3]
```diff
@@ -12,7 +13,7 @@ const comparePolicyIdCanonically = (a: Trezor.CardanoAssetGroup, b: Trezor.Carda
   // PolicyId is always of the same length
   a.policyId > b.policyId ? 1 : -1;
 
-const tokenMapToAssetGroup = (tokenMap: Cardano.TokenMap, isMint: boolean): Trezor.CardanoAssetGroup[] => {
+const tokenMapToAssetGroup: Transform<Cardano.TokenMap, Trezor.CardanoAssetGroup[]> = (tokenMap, isMint) => {
```
Comment (rhyslbw) [inline packages/hardware-trezor/src/transformers/assets.ts:16]: Interesting, so this just needed the generic type applied

File: packages/hardware-trezor/src/transformers/auxiliaryData.ts [hunk 4]
```diff
@@ -1,6 +1,10 @@
...
 import * as Trezor from '@trezor/connect';
+import { Transform } from '@cardano-sdk/util';
 
-export const mapAuxiliaryData = (auxiliaryDataHash: Crypto.Hash32ByteBase16): Trezor.CardanoAuxiliaryData => ({
-  hash: auxiliaryDataHash
+export const mapAuxiliaryData: Transform<Crypto.Hash32ByteBase16, Trezor.CardanoAuxiliaryData> = (
+  auxiliaryDataHash
+) => ({
+  cVoteRegistrationParameters: undefined,
+  hash: auxiliaryDataHash // Voting is not handled for now
```
Comment (rhyslbw) [inline packages/hardware-trezor/src/transformers/auxiliaryData.ts:9]: This comment applies to line 8 right?

**Review by rhyslbw — CHANGES_REQUESTED** [review]

File: packages/hardware-trezor/src/transformers/auxiliaryData.ts (hunk 4 above)
Comment (tomislavhoracek) [reply packages/hardware-trezor/src/transformers/auxiliaryData.ts]: @rhyslbw fixed in [e69f4b6](https://github.com/input-output-hk/cardano-js-sdk/pull/1096/commits/e69f4b619fd1ee41c5d5189beea8442a30c375df)

**Comment by tomislavhoracek** [comment]
@mkazlauskas approval was dismissed because of the last [e69f4b6](https://github.com/input-output-hk/cardano-js-sdk/pull/1096/commits/e69f4b619fd1ee41c5d5189beea8442a30c375df). Could you check it please ?

**Review by mkazlauskas — APPROVED** [review]

**Review by rhyslbw — APPROVED** [review]
Well done @tomislavhoracek 🎉

**Comment by rhyslbw** [comment]
Please squash

**Comment by tomislavhoracek** [comment]
@rhyslbw squashed

**Pull request merged**